File: services/llm-gateway/src/redactor.py
from __future__ import annotations

import re
from typing import Any


PHONE_RE = re.compile(r"(?<!\d)1[3-9]\d{9}(?!\d)")
KEY_RE = re.compile(r"sk-[A-Za-z0-9]{8,}")


def redact_text(text: str) -> str:
    text = PHONE_RE.sub("[REDACTED_PHONE]", text)
    text = KEY_RE.sub("[REDACTED_API_KEY]", text)
    return text


def redact_payload(payload: Any) -> Any:
    if isinstance(payload, str):
        return redact_text(payload)
    if isinstance(payload, list):
        return [redact_payload(item) for item in payload]
    if isinstance(payload, dict):
        return {key: redact_payload(value) for key, value in payload.items()}
    return payload

File: services/llm-gateway/ts-src/redactor.ts
const PHONE_RE = /(?<!\d)1[3-9]\d{9}(?!\d)/g;
const KEY_RE = /sk-[A-Za-z0-9]{8,}/g;

export function redactText(text: string): string {
  text = text.replace(PHONE_RE, "[REDACTED_PHONE]");
  text = text.replace(KEY_RE, "[REDACTED_API_KEY]");
  return text;
}

export function redactPayload(payload: unknown): unknown {
  if (typeof payload === "string") {
    return redactText(payload);
  }
  if (Array.isArray(payload)) {
    return payload.map(redactPayload);
  }
  if (payload !== null && typeof payload === "object") {
    const result: Record<string, unknown> = {};
    for (const [key, value] of Object.entries(
      payload as Record<string, unknown>,
    )) {
      result[key] = redactPayload(value);
    }
    return result;
  }
  return payload;
}
